src/Shared/Recruit.Vacancies.Client/Infrastructure/Reports/ReportStrategyResult.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
+
+namespace Esfa.Recruit.Vacancies.Client.Infrastructure.Reports
+{
+    public class ReportStrategyResult
```

Comment:
If we pass in a strategy for creating the header-per-report type then we won't need this added complication

Reply:
The report strategy generates all of the data (both in the result set and in the header) but just identifies that some of the data is for the header section and some is for the data section.

This data is stored in the Report in the DB so is only ever calculated once.

Reply:
I was just wondering if we should have a new "render strategy" per report type which would calculate and add headers and the like at time of download. I'm wondering if the concept of a 'header' is special enough to warrant its own persisted property.  Its just speculation at the moment and as we only have 2 reports both of which require a header then this solution is fine.  :+1: